sys/kern/kern_proc.c

```diff
@@ -2683,14 +2689,21 @@ sysctl_kern_proc_c18n_compartments(SYSCTL_HANDLER_ARGS)
 		 */
 		error = proc_read_string_properly(curthread, p,
```

Comment:
Robert's failure is actually `EPROT` from his ktrace, and I bet it is inside of here, perhaps where we use `cheri_can_access`.

Reply:
Actually, there is an `EPROT` a few lines earlier here due to a `cheri_can_access` you'd want to log as well.

Reply:
Hmm, I wonder why we do a separate cheri_can_access on each entry instead of just doing a single check for the entire array of `info.comparts` up front.

Reply:
Yes, good point. We can definitely do that.